### PR TITLE
[Evented PLEG] Ignore container events with nil PodSandboxStatus

### PR DESCRIPTION
**Pull request by ruiwen-zhao**
Signed-off-by: ruiwen-zhao <ruiwen@google.com>

#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:

This is part of the effort of [Evented PLEG](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/3386-kubelet-evented-pleg/README.md)

#### Which issue(s) this PR fixes:

Fixes #

This PR fixes the case where container runtime returns a `ContainerEventResponse` with `PodSandboxStatus == nil`. Currently, if container runtime does so, kubelet will panic on a nil pointer. 

This PR ignores the `ContainerEventResponse` when `PodSandboxStatus == nil`. It is safe to do this because
a) a event would have been received for the sandbox deletion,
b) in worst case, a relist will eventually sync the pod status.

See https://github.com/containerd/containerd/pull/7073#discussion_r1042609864 for why containerd might send a `ContainerEventResponse` with `PodSandboxStatus == nil` upon container stop/deletion.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by mikebrow — COMMENTED** [review]
suggest a log entry indicating that the event was received with nil for pod status indicating the pod is no longer present for this container in the container runtime

suggest a TODO for addressing this further

**Comment by harche** [comment]
Thanks @ruiwen-zhao. 

/lgtm

**Comment by harche** [comment]
> suggest a log entry indicating that the event was received with nil for pod status indicating the pod is no longer present for this container in the container runtime
> 
> suggest a TODO for addressing this further

/lgtm cancel 

Once @mikebrow's suggestion is added in, we can add the lgtm tag back. 

Thanks.

**Review by mikebrow — APPROVED** [review]
LGTM
@harche

**Review by mrunalp — APPROVED** [review]

File: pkg/kubelet/pleg/evented.go [hunk 1]
```diff
@@ -204,6 +204,19 @@ func (e *EventedPLEG) watchEventsChannel() {
...
 	for event := range containerEventsResponseCh {
+		// Ignore the event if PodSandboxStatus is nil.
+		// This might happen under some race condition where the podSandbox has
+		// been deleted, and therefore container runtime couldn't find the
+		// podSandbox for the container when generating the event.
+		// It is safe to ignore because
+		// a) a event would have been received for the sandbox deletion,
+		// b) in worst case, a relist will eventually sync the pod status.
+		// TODO(#114371): Figure out a way to handle this case instead of ignoring.
+		if event.PodSandboxStatus == nil || event.PodSandboxStatus.Metadata == nil {
```
Comment (fuweid) [inline pkg/kubelet/pleg/evented.go:215]: Maybe we can use `event.GetPodSandboxStatus().GetMetadata() == nil`, IIUC.

**Pull request merged**